packages/deployment/app/kes.override.js

```diff
@@ -401,35 +431,55 @@ class UpdatedKes extends Kes {
         resolve(JSON.parse(body).tag_name);
       });
     });
-  };
+  }
 
   /**
    * Determines which release version should be downloaded from
    * cumulus-message-adapter repository and then downloads that file.
    *
-   * @return {Promise}
+   * @returns {Promise} returns the path of the extracted message adapter or an empty response
    */
   fetchMessageAdapter() {
+    if (!this.config.message_adapter_filename) return Promise.resolve();
+
     const messageAdapterVersion = this.config.message_adapter_version;
     const releaseDownloadBaseUrl = `https://github.com/${this.messageAdapterGitPath}/releases/download`;
     const messageAdapterFilename = this.config.message_adapter_filename;
 
+    const releaseLocation = `${releaseDownloadBaseUrl}/` +
+                            `${messageAdapterVersion}/${messageAdapterFilename}`;
+    let adapterVersion = Promise.resolve(releaseLocation);
     if (!messageAdapterVersion) {
-      return this.fetchLatestMessageAdapterRelease()
-        .then((latestReleaseVersion) => {
-          const releaseLocation = `${releaseDownloadBaseUrl}/${latestReleaseVersion}/${messageAdapterFilename}`;
-          return this.downloadZipfile(releaseLocation, messageAdapterFilename);
-        });
-    } else {
-      const releaseLocation = `${releaseDownloadBaseUrl}/${messageAdapterVersion}/${messageAdapterFilename}`;
-      return this.downloadZipfile(releaseLocation, messageAdapterFilename);
+      adapterVersion = this.fetchLatestMessageAdapterRelease()
+        .then((latestReleaseVersion) => `${releaseDownloadBaseUrl}/` +
+                                        `${latestReleaseVersion}/${messageAdapterFilename}`);
     }
-  };
+
+    // message adapter folder
+    const folderName = path.basename(messageAdapterFilename, '.zip');
+    const kesFolder = path.join(this.config.kesFolder, 'build');
+    const adapterUnzipPath = path.join(process.cwd(), kesFolder, 'adapter', folderName);
+    const adapterZipPath = path.join(process.cwd(), kesFolder, messageAdapterFilename);
+
+    return adapterVersion
+      .then((location) => this.downloadZipfile(location, adapterZipPath))
+      .then(() => this.extractZipFile(adapterZipPath, adapterUnzipPath));
+  }
+
+  /**
+   * Override CF compilation to inject Sled
+   *
+   * @returns {Promise} returns the promise of an AWS response object
+   */
+  compileCF() {
+    // return Promise.resolve();
+    return this.fetchMessageAdapter().then(super.compileCF());
```

Comment:
Should be `super.compileCF`, not `super.compileCF()`.  The argument to `then` needs to be the function, not the result of calling the function.